crates/serialize_hierarchy/src/lib.rs

```diff
@@ -63,6 +88,24 @@ where
     }
 }
 
+impl<T> SerializeHierarchy for HashSet<T> {
+    fn serialize_hierarchy(&self, field_path: &str) -> Result<Value> {
+        bail!("cannot access HashSet with path: {}", field_path)
+    }
+
+    fn deserialize_hierarchy(&mut self, field_path: &str, _data: Value) -> Result<()> {
+        bail!("cannot access HashSet with path: {}", field_path)
+    }
+
+    fn exists(_field_path: &str) -> bool {
+        true
+    }
```

Comment:
`exists = true` is our default for all primitive types. Only `Option<T>` is properly supported. At least it fails somewhere and the `exists` catch is only a little earlier than the `serialize_hierarchy` call (so we don't gain much). I think it does not matter that much and since it would be more effort to implement a sensible checker based on the `field_path`, I would like to keep it as dumb as it is.

Reply:
You are right. We cannot traverse into a primitive type. If we get called with a type, the primitive type would need to support some kind of fields. Since this is a flat type, this cannot work and we should not advertise the opposite. I will change this.

Reply:
Done